compiler/modules/front.syntax/parseStmt.bal

```diff
@@ -76,28 +76,29 @@ function parseStmt(Tokenizer tok) returns Stmt|err:Syntax {
 }
 
 
-function finishIdentifierStmt(Tokenizer tok, string identifier, Position pos) returns Stmt|err:Syntax {
+function finishIdentifierStmt(Tokenizer tok, string identifier, Position pos, string? prefix = ()) returns Stmt|err:Syntax {
```

Comment:
The subset grammar won't allow most of these cases because they don't make sense semantically in the subset, because we don't have module-level variables and constants are not structured. The handling for `:` also makes sense only where there's not already a prefix. I think a better approach is to have two separate functions and then factor out the logic for common cases. We should really have the grammar first and then write the code to match the grammar.